noxfile.py

```diff
@@ -195,27 +241,55 @@ def build_release(session):
 
     session.log("# Ensure no files in dist/")
     if release.have_files_in_folder("dist"):
-        session.error("There are files in dist/. Remove them and try again")
+        session.error(
+            "There are files in dist/. Remove them and try again. "
+            "You can use `git clean -fxdi -- dist` command to do this"
+        )
 
     session.log("# Install dependencies")
     session.install("setuptools", "wheel", "twine")
 
-    session.log("# Checkout the tag")
-    session.run("git", "checkout", version, external=True, silent=True)
+    with mk_tmp_git_checkout(session, version) as build_dir_path:
+        with workdir(session, build_dir_path):
+            build_dists(session)
+
+        tmp_dist_dir = build_dir_path / 'dist'
+        session.log(f"# Copying dists from tmp {tmp_dist_dir}")
+        shutil.copytree(tmp_dist_dir, 'dist')
 
+
+def build_dists(session):
```

Comment:
If we're calling it `build_dists`, this should definitely not contain twine check, or anything other than the setup.py call.

I'm in favor of not actually making this function, but instead indenting the setup.py calls in accordance to `build_release -> isolated_temporary_checkout -> workdir` and leaving the rest of the code as-is.

Reply:
Not sure I understand what you want here. My main motivation is to avoid highly indented code blocks. Also, additionally, it's nice to label blocks of code with function names which make good use of abstractions.

Reply:
Oh, and also `twine check` is inside because I wanted to do the validation *before* copying the dists back to the main work dir...

Reply:
*(marking as resolved for now, after the discussion in private)*